src/bm_sim/checksums.cpp

```diff
@@ -146,10 +148,21 @@ CalcBasedChecksum::update_(Packet *pkt) const {
   f_cksum.set(cksum);
 }
 
+std::string
+CalcBasedChecksum::convert(uint64_t val) const {
+    std::stringstream ss;
+    ss << std::hex << val;
+    return ("0x" + ss.str());
+}
+
 bool
 CalcBasedChecksum::verify_(const Packet &pkt) const {
   const uint64_t cksum = calculation->output(pkt);
   const auto &f_cksum = pkt.get_phv()->get_field(header_id, field_offset);
+#ifdef BM_LOG_DEBUG_ON
```

Comment:
this `#ifdef` guard is not needed. If it is not defined, `BMLOG_DEBUG_PKT` will be a no-op and parameters won't be evaluated.